src/main/Application.h

```diff
@@ -249,6 +249,8 @@ class Application
     // instances
     virtual Hash const& getNetworkID() const = 0;
 
+    virtual void validateNetworkPassphrase() = 0;
```

Comment:
I don't think we care about this in the public interface: it's an implementation detail of validations performed by the application on startup

Reply:
I agree. I have moved this check into a non-member function just to conceptually separate it from the other code in `Application::create`. Depending on how firmly you believe that `Application` should be just an interface with no concrete implementations, there are other options such as private member or private static. Let me know if you think one of these alternatives is better.